core/controllers/topic_editor.py

```diff
@@ -364,6 +365,26 @@ def get(self, topic_id):
 
         self.render_json(self.values)
 
+class TopicPublishSendMailHandler(base.BaseHandler):
+    """A handler for sending mail to admins to review and publish topic."""
+
+    GET_HANDLER_ERROR_RETURN_TYPE = feconf.HANDLER_TYPE_JSON
+
+    @acl_decorators.can_view_any_topic_editor
+    def put(self, topic_id):
+        """Returns the TopicRights object of a topic."""
+        topic_domain.Topic.require_valid_topic_id(topic_id)
```

Comment:
Just to check ... should this be covered by the can_view_any_topic_editor decorator? Not sure.

Reply:
Yeah, it could be either `can_edit_topic` or `can_view_any_topic_editor`. It depends on whether we want to allow any topic manager to request a topic to be published or just those assigned to a topic.
I thought since topic managers are manually assigned roles in the first place, maybe this could be done by anyone. What do you think?

Reply:
Oh, sorry, I wasn't clear. I meant, should this line (line 376) be part of the can_view_any_topic_editor decorator, rather than part of this put() method?

Reply:
Oh, ok. Yeah, it can be in the decorator.
Done.